crypto/fipsmodule/self_check/self_check.c

```diff
@@ -1198,10 +1198,15 @@ int BORINGSSL_self_test(void) {
       // When requested to run self tests, also run the lazy tests.
       !boringssl_self_test_rsa() ||
       !boringssl_self_test_ecc() ||
-      !boringssl_self_test_ffdh()) {
+      !boringssl_self_test_ffdh() ||
+      !boringssl_self_test_sha256() ||
+      !boringssl_self_test_hmac_sha256()) {
+#if defined(BORINGSSL_FIPS)
+    return AWS_LC_FIPS_error("BORINGSSL_self_test did not pass.", ERR_R_FIPS_TEST_FAILURE);
```

Comment:
could this return 0 if not not build in fips mode? avoids the if-def things

Reply:
In non-FIPS mode this function return 0 or 1. In FIPS mode this function either:
* returns 1, or aborts if AWS_LC_fips_failure_callback is not defined
* returns 1, or calls AWS_LC_fips_failure_callback and returns 0